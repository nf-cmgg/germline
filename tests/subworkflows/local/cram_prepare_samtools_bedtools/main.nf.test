nextflow_workflow {

    name "Test Workflow CRAM_PREPARE_SAMTOOLS_BEDTOOLS"
    script "subworkflows/local/cram_prepare_samtools_bedtools/main.nf"
    workflow "CRAM_PREPARE_SAMTOOLS_BEDTOOLS"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "cram_prepare_samtools_bedtools"

    test("cram_prepare_samtools_bedtools - default - WGS") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:1],
                    file(params.cram1, checkIfExists:true),
                    file(params.crai1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:1],
                    []
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[4] = []
                input[5] = false
                """
            }
        }

        then {
            def fasta = "https://github.com/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/genome/hg38_chr21_22000000_23000000.fasta"
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.ready_crams.collect {
                        [ it[0], it[1], file(it[2]).name ]
                    },
                    workflow.out.ready_bams.collect {
                        [ it[0], "${file(it[1]).name},readsMD5:${bam(it[1]).getReadsMD5()}", file(it[2]).name ]
                    },
                    workflow.out.ready_beds,
                    workflow.out.reports
                ).match("default - WGS") }
            )
        }

    }

    test("cram_prepare_samtools_bedtools - default - WES common ROI") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:1],
                    file(params.cram1, checkIfExists:true),
                    file(params.crai1, checkIfExists:true)
                ])
                input[1] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:1],
                    []
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[4] = Channel.fromPath(params.bed, checkIfExists:true)
                input[5] = true
                """
            }
        }

        then {
            def fasta = "https://github.com/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/genome/hg38_chr21_22000000_23000000.fasta"
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.ready_crams.collect {
                        [ it[0], it[1], file(it[2]).name ]
                    },
                    workflow.out.ready_bams.collect {
                        [ it[0], "${file(it[1]).name},readsMD5:${bam(it[1]).getReadsMD5()}", file(it[2]).name ]
                    },
                    workflow.out.ready_beds,
                    workflow.out.reports
                ).match("default - WES common ROI") }
            )
        }

    }

    test("cram_prepare_samtools_bedtools - default - WES") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:1],
                    file(params.cram1, checkIfExists:true),
                    []
                ])
                input[1] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:1],
                    file(params.bed, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[4] = []
                input[5] = false
                """
            }
        }

        then {
            def fasta = "https://github.com/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/genome/hg38_chr21_22000000_23000000.fasta"
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.ready_crams.collect {
                        [ it[0], it[1], file(it[2]).name ]
                    },
                    workflow.out.ready_bams.collect {
                        [ it[0], "${file(it[1]).name},readsMD5:${bam(it[1]).getReadsMD5()}", file(it[2]).name ]
                    },
                    workflow.out.ready_beds,
                    workflow.out.reports
                ).match("default - WES") }
            )
        }

    }

    test("cram_prepare_samtools_bedtools - default - merge") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:2],
                    file(params.cram1, checkIfExists:true),
                    []
                ],[
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:2],
                    file(params.cram2, checkIfExists:true),
                    []
                ])
                input[1] = Channel.of([
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:2],
                    file(params.bed, checkIfExists:true)
                ],[
                    [id:"NA24143", sample:"NA24143", family:"Ashkenazim", family_samples:"NA24143", duplicate_count:2],
                    []
                ])
                input[2] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists:true)
                ])
                input[3] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists:true)
                ])
                input[4] = []
                input[5] = true
                """
            }
        }

        then {
            def fasta = "https://github.com/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/genome/hg38_chr21_22000000_23000000.fasta"
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.ready_crams.collect {
                        [ it[0], "${file(it[1]).name},readsMD5:${cram(it[1], fasta).getReadsMD5()}", file(it[2]).name ]
                    },
                    workflow.out.ready_bams.collect {
                        [ it[0], "${file(it[1]).name},readsMD5:${bam(it[1]).getReadsMD5()}", file(it[2]).name ]
                    },
                    workflow.out.ready_beds,
                    workflow.out.reports
                ).match("default - merge") }
            )
        }

    }

}
