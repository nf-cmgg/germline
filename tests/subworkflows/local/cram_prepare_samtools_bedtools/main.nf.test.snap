{
    "default - merge": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 2
                    },
                    "NA24143.cram,readsMD5:be28f434d6f7bcfa398488a6611d89c1",
                    "NA24143.cram.crai"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 2
                    },
                    "NA24143.bam,readsMD5:be28f434d6f7bcfa398488a6611d89c1",
                    "NA24143.bam.bai"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 2
                    },
                    "NA24143_intersect.bed:md5,b87069698afefb15282d069e56110046"
                ]
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T10:19:20.53195135"
    },
    "default - WGS": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/illumina/crams/NA24143.cram",
                    "NA24143.cram.crai"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "NA24143.filter.bed:md5,85a5568a6976ed455caa712991b30ac2"
                ]
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T10:17:48.433466279"
    },
    "default - WES": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/illumina/crams/NA24143.cram",
                    "NA24143.cram.crai"
                ]
            ],
            [
                
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "NA24143_intersect.bed:md5,b87069698afefb15282d069e56110046"
                ]
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T10:18:36.993186258"
    },
    "default - WES common ROI": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "/nf-cmgg/test-datasets/raw/germline/data/genomics/homo_sapiens/illumina/crams/NA24143.cram",
                    "NA24143.cram.crai"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "NA24143.bam,readsMD5:77afffb023e537869c5c6ebf31187ded",
                    "NA24143.bam.bai"
                ]
            ],
            [
                [
                    {
                        "id": "NA24143",
                        "sample": "NA24143",
                        "family": "Ashkenazim",
                        "family_samples": "NA24143",
                        "duplicate_count": 1
                    },
                    "NA24143_intersect.bed:md5,b87069698afefb15282d069e56110046"
                ]
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-23T10:18:13.865281894"
    }
}